src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -927,74 +963,77 @@ def geometric_basis(G, E, EC0, p, dual_graph) -> list:
-          A vertex at (2, -2),
-          A vertex at (-2, -2)],
-         [A vertex at (-2, -2),
-          A vertex at (2, -2),
-          A vertex at (1/2, -1/2),
-          A vertex at (1/2, 1/2),
-          A vertex at (-1/2, 1/2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (1/2, -1/2),
-          A vertex at (2, -2),
-          A vertex at (-2, -2)],
-         [A vertex at (-2, -2),
-          A vertex at (2, -2),
-          A vertex at (1/2, -1/2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (-2, -2)],
-         [A vertex at (-2, -2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (-1/2, 1/2),
-          A vertex at (1/2, 1/2),
-          A vertex at (2, 2),
-          A vertex at (-2, 2),
-          A vertex at (-1/2, 1/2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (-2, -2)],
-         [A vertex at (-2, -2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (-1/2, 1/2),
-          A vertex at (-2, 2),
-          A vertex at (-2, -2)]]
+        sage: from sage.schemes.curves.zariski_vankampen import geometric_basis, corrected_voronoi_diagram, voronoi_cells
+        sage: points = (0, -1, I, 1, -I)
+        sage: V = corrected_voronoi_diagram(points)
+        sage: G, E, p, EC, DG, VR = voronoi_cells(V, vertical_lines=frozenset((0 .. 4)))
+        sage: gb, vd = geometric_basis(G, E, EC, p, DG, vertical_regions=VR)
+        sage: gb
+        [[A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (-5/2, 5/2),
+          A vertex at (-1/2, 1/2), A vertex at (-1/2, -1/2), A vertex at (1/2, -1/2),
+          A vertex at (1/2, 1/2), A vertex at (-1/2, 1/2), A vertex at (-5/2, 5/2),
+          A vertex at (5/2, 5/2), A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (-5/2, 5/2),
+          A vertex at (-1/2, 1/2), A vertex at (1/2, 1/2), A vertex at (5/2, 5/2),
+          A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (1/2, 1/2),
+          A vertex at (1/2, -1/2), A vertex at (5/2, -5/2)], [A vertex at (5/2, -5/2),
+          A vertex at (1/2, -1/2), A vertex at (-1/2, -1/2), A vertex at (-1/2, 1/2),
+          A vertex at (-5/2, 5/2), A vertex at (-5/2, -5/2), A vertex at (-1/2, -1/2),
+          A vertex at (1/2, -1/2), A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (1/2, -1/2), A vertex at (-1/2, -1/2),
+          A vertex at (-5/2, -5/2), A vertex at (5/2, -5/2)]]
+        sage: vd
+        {0: 0, 1: 3, 2: 1, 3: 2, 4: 4}
     """
     i = EC0.index(p)
-    EC = EC0[i:-1] + EC0[:i + 1]   # A counterclockwise eulerian circuit on the boundary, starting and ending at p
+    EC = EC0[i:-1] + EC0[:i + 1]
+    # A counterclockwise eulerian circuit on the boundary,
+    # starting and ending at p
     if G.size() == E.size():
         if E.is_cycle():
-            return [EC]
+            j = list(dual_graph.vertices())[0][0]
```

Comment:
```suggestion
            j = next(dual_graph.vertex_iterator())[0]
```
This should simplify things and be much more memory efficient.